Source/SwiftLintBuiltInRules/Rules/Style/NonFinalClassRule.swift

```diff
@@ -0,0 +1,51 @@
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+@SwiftSyntaxRule(correctable: true, optIn: true)
+struct NonFinalClassRule: Rule {
+    var configuration = SeverityConfiguration<Self>(.warning)
+
+    static let description = RuleDescription(
+        identifier: "non_final_class",
+        name: "Non-Final Class",
+        description: "Classes should be marked as `final` unless they are explicitly `open`.",
+        kind: .idiomatic,
+        nonTriggeringExamples: [
+            Example("final class MyClass {}"),
+            Example("open class MyClass {}"),
+            Example("public final class MyClass {}"),
+        ],
+        triggeringExamples: [
+            Example("class MyClass {}"),
+            Example("public class MyClass {}"),
+        ],
+        corrections: [
+            Example("class MyClass {}"): Example("final class MyClass {}"),
+            Example("public class MyClass {}"): Example("public final class MyClass {}"),
+        ]
+    )
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor<ConfigurationType>? {
+        Visitor(configuration: configuration, file: file)
+    }
+
+    private final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
+            let modifiers = node.modifiers
+            if !modifiers.contains(keyword: .final),
+               !modifiers.contains(keyword: .open) {
+                let classToken = node.classKeyword
+                violations.append(.init(
+                    position: classToken.positionAfterSkippingLeadingTrivia,
+                    reason: "Classes should be marked as `final` unless they are explicitly `open`",
```

Comment:
Could be more succinct and targeted to the single class being checked:
```suggestion
                    reason: "Class should be marked as `final` or explicitly `open`",
```